archunit/src/test/java/com/tngtech/archunit/core/importer/ImportOptionsTest.java

```diff
@@ -157,6 +161,30 @@ public void detects_archives_correctly(ImportOption doNotIncludeArchives) {
                 .isFalse();
     }
 
+    @DataProvider
+    public static Object[][] do_not_include_package_info_classes() {
+        return testForEach(new DoNotIncludePackageInfoClasses(), DO_NOT_INCLUDE_PACKAGE_INFO_CLASSES);
+    }
+
+    @Test
+    @UseDataProvider("do_not_include_package_info_classes")
+    public void detect_package_info_class(ImportOption doNotIncludePackageInfoClasses) {
+        NormalizedUri mockUri = mock(NormalizedUri.class);
```

Comment:
I would not mock things like this. I think mocking is great for collaborators, but not great for domain objects (you lose intrinsic invariants, i.e. you can create unrealistic objects, etc., and the tests are harder to maintain on refactorings).
Is there any reason you don't simply use something like

```
Location location = Location.of(URI.create("file:///path/to/package/and/file/package-info.class"));
```

Then the whole extra class `StubLocation` is also obsolete... :man_shrugging:

Reply:
that's funny - at the start, I actually did something similar to this and I changed it to the stubbing class instead of including an extra file.
but you are right - it is better like that.
fixed in https://github.com/TNG/ArchUnit/pull/794/commits/90c126f6acba68e0602e316c5d576001ee5f20ce